### ISSUE TITLE
[BERT/Pytorch] NaN found when pretraining BERT-Large using scripts/run_pretraining.sh

### ISSUE THREAD

**Issue by mx8435** [NVIDIA/DeepLearningExamples#1135]
**Environment**
Please provide at least:
* Container version (e.g. pytorch:19.05-py3): nvcr.io/nvidia/pytorch:21.11-py3
* GPUs in the system: (e.g. 8x Tesla V100-SXM2-16GB): 8x A100 80G
* CUDA driver version (e.g. 418.67):  470.82.01
* CUDA version: cuda_11.5.r11.5/compiler.30411180_0

**Describe the bug**
I‘ve follow the instruction of`PyTorch/LanguageModeling/BERT/README.md` and use `create_datasets_from_start.sh` to generate training data for Wikipedia and Bookcorpus, and run `run_pretraining.sh` without modify any hyper parameter.  As shown in  the following Figure, an NaN loss was found at Iteration 5310 in Phase1. I'm confused how to solve this problem. Thanks.

![image](https://user-images.githubusercontent.com/8417466/169762711-4b922730-2b89-4db1-9da6-8ae6f4894877.png)

The hyperparameters used:

train_batch_size=${1:-8192}
learning_rate=${2:-"6e-3"}
precision=${3:-"fp16"}
num_gpus=${4:-$(nvidia-smi -L | wc -l)}
warmup_proportion=${5:-"0.2843"}
train_steps=${6:-7038}
save_checkpoint_steps=${7:-200}
resume_training=${8:-"false"}
create_logfile=${9:-"true"}
accumulate_gradients=${10:-"true"}
gradient_accumulation_steps=${11:-32}
seed=${12:-12439}
job_name=${13:-"bert_lamb_pretraining"}
allreduce_post_accumulation=${14:-"true"}
allreduce_post_accumulation_fp16=${15:-"true"}
train_batch_size_phase2=${16:-4096}
learning_rate_phase2=${17:-"4e-3"}
warmup_proportion_phase2=${18:-"0.128"}
train_steps_phase2=${19:-1563}
gradient_accumulation_steps_phase2=${20:-512}
#change this for other datasets
DATASET=pretrain2/phase1/unbinned/parquet
DATA_DIR_PHASE1=${21:-$BERT_PREP_WORKING_DIR/${DATASET}/}
DATA_DIR_PHASE1=${DATASET}
#change this for other datasets
DATASET2=pretrain2/phase2/bin_size_64/parquet
DATA_DIR_PHASE2=${22:-$BERT_PREP_WORKING_DIR/${DATASET2}/}
DATA_DIR_PHASE2=${DATASET2}
CODEDIR=${23:-"../"}
init_checkpoint=${24:-"None"}
VOCAB_FILE=vocab/en_vocab.txt
RESULTS_DIR=$CODEDIR/results
CHECKPOINTS_DIR=$RESULTS_DIR/checkpoints
wikipedia_source=${25:-$BERT_PREP_WORKING_DIR/wikipedia/source/}
wikipedia_source=corpus/
num_dask_workers=${26:-$(nproc)}
num_dask_workers=64
num_shards_per_worker=${27:-128}
num_workers=${28:-4}
num_nodes=1
sample_ratio=${29:-0.9}
phase2_bin_size=${30:-64}
masking=${31:-static}

**Comment by sharathts** [comment]
It looks like an unlucky divergent run. Can you try with a different seed ?

**Comment by mx8435** [comment]
> It looks like an unlucky divergent run. Can you try with a different seed ?
@sharathts Thanks for your reply.

I've tried use seed=42 and retrain from scratch, but overflow happens at step 4691. 


![image](https://user-images.githubusercontent.com/8417466/169959649-9939391c-d5fa-4c6b-98b6-a3b333b81510.png)

**Comment by mx8435** [comment]
@sharathts I tried to turn off fp16 by setting percision=fp32, and nan happened after a new train step 2440.

![image](https://user-images.githubusercontent.com/8417466/170164679-28f71ad9-62a5-4168-8f95-876347b4f4ce.png)

**Comment by sharathts** [comment]
In that case, I would recommend lowering the learning rate to either 5.5e-3 or 5e-3

**Comment by sepzjh** [comment]
hello，How do you download your pretraining dataset, and what does the directory look like after downloading? I've been unable to download the dataset successfully.
